src/components/mols/CreateCoupon/index.tsx

```diff
@@ -0,0 +1,226 @@
+              className="w-32 !text-3xl py-3 flex items-center justify-center font-semibold rounded-full border-[1px] border-black"
+              label={values.discount + "%"}
+              errorMessage={errors.discount}
+              invalid={!!(errors.discount && touched.discount)}
+            >
+              <Field
+                invalid={!!(errors.discount && touched.discount)}
+                className="accent-primary-main !focus:border-none !hover:border-none focus:ring-0"
+                name="discount"
+                min="5"
+                max="95"
+                step="1"
+                type="range"
+                label="discount"
+                component={Input}
+              />
+            </FormItem>
+            <FormItem
+              label="Título do cupom"
+              errorMessage={errors.title}
+              invalid={!!(errors.title && touched.title)}
+            >
+              {!!isValidating && <ErrorMessage name="title" />}
+              <Field
+                invalid={!!(errors.title && touched.title)}
+                name="title"
+                type="text"
+                label="title"
+                component={Input}
+                className="bg-white"
+              />
+            </FormItem>
+            <div className="grid grid-cols-2 w-full">
+              <FormItem
+                label="Limite de cupons"
+                errorMessage={!couponsIlimited && errors.coupon_limit}
+                invalid={
+                  !couponsIlimited &&
+                  !!(errors.coupon_limit && touched.coupon_limit)
+                }
+              >
+                <Field
+                  disabled={couponsIlimited}
+                  invalid={
+                    !couponsIlimited &&
+                    !!(errors.coupon_limit && touched.coupon_limit)
+                  }
+                  name="coupon_limit"
+                  value={
+                    couponsIlimited
+                      ? (values.coupon_limit = "ilimitado")
+                      : values.coupon_limit == "ilimitado"
+                      ? (values.coupon_limit = "")
+                      : values.coupon_limit
+                  }
+                  type="text"
+                  label="coupon_limit"
+                  component={Input}
+                  className="bg-white disabled:bg-white"
+                />
+              </FormItem>
+              <FormItem
+                label="Limite por usuário"
+                errorMessage={!ilimitedByUser && errors.user_limit}
+                invalid={
+                  !ilimitedByUser && !!(errors.user_limit && touched.user_limit)
+                }
+              >
+                <Field
+                  disabled={ilimitedByUser}
+                  invalid={
+                    !ilimitedByUser &&
+                    !!(errors.user_limit && touched.user_limit)
+                  }
+                  name="user_limit"
+                  value={
```

Comment:
Add multiples ternary verification has a bad practice because the code so difficult to read. Has a other way to make this?

Reply:
We can talk about this tomorrow